### ISSUE TITLE
Build php7 extension on windows thread-safe

### ISSUE THREAD

**Issue by AlexanderGabriel** [swig/swig#1236]
Hi,

i try to build a php7-extension on windows.
To build php, i used https://github.com/Microsoft/php-sdk-binary-tools and Visual Studio 2017 Community Edition.
If i build php non-thread-safe, it works but if i build php thread-safe to use it as apache module, i get errors when compiling the module. The code uses (in my example-case) example_globals which is not defined, when php is compiled thread-safe:
![grafik](https://user-images.githubusercontent.com/23343388/39016221-0c28b6a8-4420-11e8-9d43-93ae7e486ee4.png)

I use cmake to build everything. 
Here are my files.
[CMakeLists.txt](https://github.com/swig/swig/files/1929706/CMakeLists.txt)
[example.c.txt](https://github.com/swig/swig/files/1929650/example.c.txt)
[example.i.txt](https://github.com/swig/swig/files/1929651/example.i.txt)

To test, rename example.c.txt to example.c and example.i.txt to example.i

Building against non-thread-safe php works (change the comments in CMakeLists.txt):
![grafik](https://user-images.githubusercontent.com/23343388/39017040-79efbb58-4422-11e8-9000-e67843573dcf.png)

Is this a problem with swig or is it not designed to work in thread-safe environments or am i just doing anything wrong?

Thank you.

**Comment by ojwb** [comment]
It'd be more helpful to cut and paste the errors as text.

SWIG-generated wrappers should work with threads, and in fact I thought PHP7 had done away with the PHP-specific ZTS thread mechanisms that older versions had so I'd expect it to work with fewer issues than with PHP5.

I'm afraid it's not a platform I use so you'll probably need to investigate for yourself.

**Comment by ClosetMonkey** [comment]
@ojwb No, php7 still uses ZTS. Only difference where extension development is concerned is that the php7 php_api no longer requires `tsrm_ls` pointers in all of it's functions.

**Comment by ojwb** [comment]
@ClosetMonkey The `tsrm_ls` pointers and associated macros are what I meant by "the PHP-specific ZTS thread mechanisms that older versions had".  It's not useful to be adding those macros to the code we generate for PHP7 as was being proposed here - the PHP7 C API headers define them still, but only as no-op versions.

The bug here turned out to be something that had already been fixed in git by 224bb9e0232a219b7cefd23472b592b39e277f46.

**Comment by AlexanderGabriel** [comment]
The tsrm_ls stuff needs to stay there to keep php5-compatibility. I tried to remove it white working on https://github.com/mapserver/mapserver/pull/5687 but this will break php5.

**Comment by AlexanderGabriel** [comment]
Oh guys... I'm sorry. I thought this thread was a issue on mapserver-project. too early in the morning. sorry.
I deleted my comment.
